Administrator Guide/Trash.md

```diff
@@ -0,0 +1,80 @@
+
+2. *gluster volume set &lt;VOLNAME> features.trash-dir &lt;name>*
+
+    This command is used to reconfigure the trash directory to a user specified name. The argument is a valid directory name. Directory will be created inside every brick under this name. If not specified by the user, the trash translator will create the trash directory with the default name “.trashcan”. This can be used only when trash-translator is on.
+
+3. *gluster volume set &lt;VOLNAME> features.trash-max-filesize &lt;size>*
+
+    This command can be used to filter files entering trash directory based on their size. Files above trash_max_filesize are deleted/truncated directly. Value for size may be followed by mutliplicative suffixes KB (=1024), MB (=1024*1024 and GB. Default size is set to 5MB. As of now any value specified higher than 1GB will be changed to 1GB at the maximum level.
+
+4. *gluster volume set &lt;VOLNAME> features.trash-eliminate-path &lt;path1> [ , &lt;path2> , . . . ]*
+
+    This command can be used to set the eliminate pattern for the trash translator. Files residing under this pattern will not be moved to trash directory during deletion/truncation. Path must be a valid one present in volume.
+
+5. *gluster volume set &lt;VOLNAME> features.trash-internal-op &lt;on | off>*
+
+    This command can be used to enable trash for internal operations like self-heal and re-balance. By default set to off.
+
+##Testing
+Following steps give illustrates a simple scenario of deletion of file from directory
+
+1. Create a distributed volume with two bricks and start it.
+
+        gluster volume create test rhs:/home/brick
+
+        gluster volume start test
+
+2. Enable trash translator
+
+        gluster volume set test feature.trash on
+
+3. Mount glusterfs client  as follows.
+
+        mount -t glusterfs  rhs:test /mnt
+
+4. Create a directory and file in the mount.
+
+        mkdir mnt/dir
+
+        echo abc > mnt/dir/file
+
+5. Delete the file from the mount.
+
+        rm mnt/dir/file -rf
+
+6. Checkout inside the trash directory.
+
+        ls mnt/.trashcan
+
+We can find the deleted file inside the trash directory with timestamp appending on its filename.
+
+For example,
+
+        [root@rh-host ~]#mount -t glusterfs rh-host:/test /mnt/test
+        [root@rh-host ~]#mkdir /mnt/test/abc
+        [root@rh-host ~]#touch /mnt/test/abc/file
+        [root@rh-host ~]#rm /mnt/test/abc/filer
+        remove regular empty file ‘/mnt/test/abc/file’? y
+        [root@rh-host ~]#ls /mnt/test/abc
+        [root@rh-host ~]#
+        [root@rh-host ~]#ls /mnt/test/.trashcan/abc/
+        file2014-08-21_123400
+
+##Points to be remembered
+[1] As soon as the volume is started, trash directory will be created inside the volume and will be visible through mount. Disabling trash will not have any impact on its visibilty from the mount.
```

Comment:
This list is not rendered properly. Please test it with a markdown renderer such as [http://markable.in/editor/](http://markable.in/editor/ )